catx/network_module.py

```diff
@@ -0,0 +1,40 @@
+from abc import abstractmethod
+import haiku as hk
+from chex import PRNGKey
+from catx.type_defs import Observations, NetworkExtras, Logits
+
+
+class CustomHaikuNetwork(hk.Module):
```

Comment:
I am personally not a fan of this name. We already know it's custom. We probably want a name that tells us where this network is used (inside CATS tree) and what it's inheriting (hk.Module).
Some suggestions: `CATSHaikuNetwork` or `CATXHaikuNetwork` or `CATXNetwork` or something else.